app-tasks/dags/geotiff/find_geotiff_scenes.py

```diff
@@ -68,11 +60,15 @@ def find_geotiffs(*args, **kwargs):
         )
         logger.info('Kicking off new scene import: %s', run_id)
         upload = Upload.from_id(upload_id)
-        upload.update_upload_status('Queued')
-        conf['upload_id'] = upload_id
-        confjson = json.dumps(conf)
-        dag_args = DagArgs(dag_id=dag_id, conf=confjson, run_id=run_id)
-        trigger_dag(dag_args)
+        # We already tried to check this above but the uploads endpoint doesn't currently
+        # support an uploadStatus filter so we have to check again to avoid trying to
+        # kick off imports and setting the status a second time
+        if upload.uploadStatus.upper() == 'UPLOADED':
+            upload.update_upload_status('Queued')
+            conf['upload_id'] = upload_id
+            confjson = json.dumps(conf)
+            dag_args = DagArgs(dag_id=dag_id, conf=confjson, run_id=run_id)
+            trigger_dag(dag_args)
```

Comment:
It'd be good to put everything inside the `if` block in a `try`/`except` where we set the status to `Failed` and then re-raise, similarly to what is happening in `import_geotiff_scenes`.